base/bucket_gocb.go

```diff
@@ -2073,6 +2074,40 @@ func (bucket CouchbaseBucketGoCB) Close() {
 	}
 }
 
+
+
+func (bucket CouchbaseBucketGoCB) GetExpiry(k string) (expiry uint32, getMetaError error) {
+
+	bucket.singleOps <- struct{}{}
+	gocbExpvars.Add("SingleOps", 1)
+	defer func() {
+		<-bucket.singleOps
+		gocbExpvars.Add("SingleOps", -1)
+	}()
+
+	agent := bucket.IoRouter()
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	getMetaCallback := func(value []byte, flags uint32, cas gocbcore.Cas, exp uint32, seq gocbcore.SeqNo, dataType uint8, deleted uint32, err error) {
+		defer wg.Done()
+		if err != nil {
+			getMetaError = pkgerrors.Wrapf(getMetaError, "Error getting expiry for doc: %s", k)
```

Comment:
Can we add support for retry on temporary server errors, similar to other gocb operations?

Reply:
I renamed this to `getExpirySingleAttempt()` and added a retry wrapper `GetExpiry()`.

The first attempt to do it all in one function ended up with nested closure functions which was pretty hard to understand.